nuxeo-features/nuxeo-platform-imaging/nuxeo-platform-imaging-core/src/main/java/org/nuxeo/ecm/platform/picture/listener/PictureChangedListener.java

```diff
@@ -78,8 +78,8 @@ public void handleEvent(Event event) {
             Property viewsProp = doc.getProperty(AbstractPictureAdapter.VIEWS_PROPERTY);
 
             Boolean forceGeneration = Boolean.TRUE.equals(doc.getContextData(CTX_FORCE_VIEWS_GENERATION));
-            if (forceGeneration
-                    || !viewsProp.isDirty() && (ABOUT_TO_CREATE.equals(event.getName()) || fileProp.isDirty())) {
+            if (forceGeneration || (!viewsProp.isDirty() || viewsProp.size() == 0)
+                    && (ABOUT_TO_CREATE.equals(event.getName()) || fileProp.isDirty())) {
```

Comment:
The parenthesing and mix of `&&` and `||` don't make it easy to understand the logic being tested. I'd rather see something like:

            boolean noPictureViews = !viewsProp.isDirty() || viewsProp.size() == 0;
            boolean fileChanged = ABOUT_TO_CREATE.equals(event.getName()) || fileProp.isDirty();
            if (forceGeneration || (noPictureViews && fileChanged)) {

(Change the names as you see fit)